spec/Section 6 -- Execution.md

```diff
@@ -616,30 +712,84 @@ must only allow usage of variables of appropriate types.
+  * Create {generator} from {ResolveFieldValue(objectType, objectValue, fieldName, argumentValues)}.
+  * Return {generator}.
 
 Note: It is common for {resolver} to be asynchronous due to relying on reading
 an underlying database or networked service to produce a value. This
 necessitates the rest of a GraphQL executor to handle an asynchronous
-execution flow.
-
+execution flow. In addition, a commom implementation of {generator} is to leverage 
+asynchronous iterators or asynchronous generators provided by many programming languages.
 
 ### Value Completion
 
 After resolving the value for a field, it is completed by ensuring it adheres
 to the expected return type. If the return type is another Object type, then
-the field execution process continues recursively.
-
-CompleteValue(fieldType, fields, result, variableValues):
+the field execution process continues recursively. In the case where a value
+returned for a list type field is an iterator due to `@stream` specified on the 
+field, value completition iterates over the iterator until the number of items
+yield by the iterator satisfies `initialCount` specified on the `@stream` directive.
+Unresolved items in the iterator will be stored in a stream record which the executor
+resumes to execute after the initial execution finishes.
+
+#### Stream Record
+
+**Formal Specification**
+Let {streamField} be a list field with a `@stream` directive provided.
+A Stream Record is a structure containing:
+  {label}: value derived from the `@stream` directive's `label` argument.
+  {iterator}: created by {ResolveFieldGenerator}.
+  {resolvedItems}: items resolved from the {iterator} but not yet delivered.
+  {index}: indicating the position of the item in the complete list.
+  {path}: a list of field names and indices from root to {streamField}.
+  {fields}: the group of fields grouped by CollectFields() for {streamField}.
+  {innerType}: inner type of {streamField}'s type.
+
+CreateStreamRecord(label, initialCount, iterator, resolvedItems, index, fields, innerType):
+*  Construct a stream record based on the parameters passed in.
+
+ResolveStreamRecord(streamRecord, variableValues, subsequentPayloads):
+  * Let {label}, {iterator}, {resolvedItems}, {index}, {path}, {fields},
+    {innerType} be the correspondent fields on the Stream Record structure.
+  * Remove the first entry from {resolvedItem}, let the entry be {item}. If 
+    {resolvedItem} is empty, retrieve more items from {iterator}:
+    * Append {index} to {path}.
+    * Increment {index}.
+    * Let {payload} be the result of calling CompleteValue(innerType, fields, item, variableValues, subsequentPayloads, path)}.
+    * Add an entry to {payload} named `label` with the value {label}.
+    * Add an entry to {payload} named `path` with the value {path}.
+    * If {resolveItem} is not empty or {iterator} does not reach the end:
+      * Append {streamRecord} to {subsequentPayloads}.
+    * Return {payload}.
```

Comment:
I think you might need to revisit this definition; I might be reading it wrong but it feels like you may have used `resolvedItem` in the place of `resolvedItems` in one place, and the "retrieve more items from {iterator}" isn't super clear, it refers to "If {resolvedItem} is not empty" under an item which already seems to state "If {resolvedItem} is empty", and there doesn't seem to be a point at which `{resolvedItem}` would change between the two. Also it uses `resolveItem` rather than `resolvedItem` which I think is a typo since it doesn't seem to be referred to elsewhere.